File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.entity.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {

    /**
     * 查询所有品牌列表
     * @return
     */
    List<TbBrand> findAll();

    /**
     * 查询分页所有品牌列表
     * @param pageNum 当前页码
     * @param pageSize 每页条数
     * @return
     */
    PageResult findPage(int pageNum, int pageSize);

    /**
     * 根据条件分页查询
     * @param tbBrand
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult search(TbBrand tbBrand, int pageNum, int pageSize);
    /**
     * 添加品牌
     * @param tbBrand
     */
    void save(TbBrand tbBrand);

    /**
     * 根据id查询品牌
     * @param id
     * @return
     */
    TbBrand findOne(Long id);

    /**
     * 更新品牌信息
     * @param tbBrand
     */
    void update(TbBrand tbBrand);

    /**
     * 批量删除品牌
     * @param ids
     */
    void delete(long[] ids);

    /**
     * 品牌下拉框数据
     * @return
     */
    List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//服务层
app.service("brandService", function ($http) {
    //查询所有品牌列表
    this.findAll=function () {
        return $http.get("../brand/findAll.do");
    }
    //分页查询所有品牌列表
    this.findPage=function (page, size) {
        return $http.get("../brand/findPage.do?page="+page+"&size="+size);
    }
    //条件分页查询
    this.search=function (page, size, searchEntity) {
        return $http.post("../brand/search.do?page="+page+"&size="+size, searchEntity);
    }
    //根据id查询
    this.findOne=function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    }
    //保存品牌
    //1.没有id保存
    this.save=function (entity) {
        return $http.post("../brand/save.do", entity);
    }
    //2.有id更新
    this.update=function (entity) {
        return $http.post("../brand/update.do", entity);
    }
    //批量删除
    this.delete=function (ids) {
        return $http.get("../brand/delete.do?ids="+ids);
    }
    //下拉列表数据
    this.selectOptionList=function(){
        return $http.get('../brand/selectOptionList.do');
    }
});